src/policy_artifacthub.rs

```diff
@@ -0,0 +1,325 @@
+use anyhow::{anyhow, Result};
+use csv::ReaderBuilder;
+use semver::Version;
+use serde::{Deserialize, Serialize};
+use std::collections::HashSet;
+use time::OffsetDateTime;
+use url::Url;
+
+use crate::policy_metadata::{self, MetadataV2};
+
+/// Format of artifacthub-pkg.yml file as defined in
+/// https://github.com/artifacthub/hub/blob/master/docs/metadata/artifacthub-pkg.yml
+#[derive(Deserialize, Serialize, Debug, Clone)]
+#[serde(rename_all = "camelCase")]
+pub struct ArtifactHubPkg {
+    pub version: Version,
+    pub name: String,
+    pub display_name: String,
+    #[serde(with = "time::serde::rfc3339")]
+    pub created_at: OffsetDateTime,
+    pub description: String,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub license: Option<String>,
+    #[serde(rename = "homeURL", skip_serializing_if = "Option::is_none")]
+    pub home_url: Option<Url>, // https://github.com/kubewarden/verify-image-signatures
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub containers_images: Option<Vec<ContainerImage>>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub keywords: Option<Vec<String>>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub links: Option<Vec<Link>>,
+    // #[serde(skip_serializing_if = "Option::is_none")]
+    // pub maintainers: Option<Vec<Maintainer>>,
+    pub provider: Provider,
+    pub recommendations: Vec<Recommendation>,
+    pub annotations: HashSet<Annotation>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub readme: Option<String>,
+}
+
+impl Default for ArtifactHubPkg {
+    fn default() -> Self {
+        let now = OffsetDateTime::now_utc();
+        Self {
+            created_at: now,
+            version: Version::new(0, 0, 0),
+            name: Default::default(),
+            display_name: Default::default(),
+            description: Default::default(),
+            license: Default::default(),
+            home_url: Default::default(),
+            containers_images: Default::default(),
+            keywords: Default::default(),
+            links: Default::default(),
+            provider: Default::default(),
+            recommendations: Default::default(),
+            annotations: Default::default(),
+            readme: Default::default(),
+        }
```

Comment:
It can't be used here :/, it applies to `Self`, and makes the assignation recursive: https://internals.rust-lang.org/t/pre-pre-rfc-the-default-default-shorthand-is-misleading/15917.

Reply:
Resolving, irrelevant after refactor.